skimage/draw/draw.py

```diff
@@ -163,7 +166,38 @@ def set_color(img, coords, color):
            [0, 0, 0, 0, 0, 0, 0, 0, 0, 1]], dtype=uint8)
 
     """
-
     rr, cc = coords
-    rr, cc = _coords_inside_image(rr, cc, img.shape)
-    img[rr, cc] = color
+
+    if alpha is None:
+        rr, cc = _coords_inside_image(rr, cc, img.shape)
+    else:
+        rr, cc, alpha = _coords_inside_image(rr, cc, img.shape, val=alpha)
+
+    if not np.isscalar(color):
+        color = np.array(color)
+        if color.shape[0] != img.shape[-1]:
+            raise ValueError('Color shape ({}) must match last '
+                             'image dimension ({}).'.format(color.shape[0],
+                                                            img.shape[-1]))
+
+    if alpha is None:
+        img[rr, cc] = color
+    else:
+        if np.isscalar(alpha) or np.isscalar(color):
+            color = color * alpha
+        else:
+            color = color * alpha[:, None]
+
+
+        # Strategy: try operation directly.  If scalar / correctly shaped
+        # vector, this will work.  If not (e.g. vector alpha but color image),
+        # try broadcasting.
+        try:
+            vals = img[rr, cc] * (1 - alpha)
+        except ValueError:
+            vals = img[rr, cc] * (1 - alpha[:, None])
```

Comment:
It seems like you are doing this pattern over and over. Is there no way to check the shape of things *once* to ensure that all the operations will work first-shot?